src/DittoMachine.sol

```diff
@@ -425,7 +425,7 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver {
                 cloneShape.tokenId,
                 cloneShape.worth
             );
-            if (royaltyAmount > 0) {
+            if (royaltyAmount > 0 && royaltyAmount < cloneShape.worth) {
                 cloneShape.worth -= royaltyAmount;
```

Comment:
this checks for over/underflow already. Don't think the extra check is necessary

Reply:
this is so that if `royaltyAmount >= cloneShape.worth`, it doesn't revert and transfers the nft without royalty payments.

Reply:
I don't remember the conclusion here. do you?

Reply:
No, I don't either. But, I'm not against this check being put here if you'd still like to add it.